firewall/privacy_mapper_test.go

```diff
@@ -302,13 +353,224 @@ func TestPrivacyMapper(t *testing.T) {
 				FailedUpdates: []*lnrpc.FailedUpdate{
 					{
 						Outpoint: &lnrpc.OutPoint{
-							TxidStr:     "097ef666a61919ff3413b3b701eae3a5cbac08f70c0ca567806e1fa6acbfe384",
-							OutputIndex: 2161781494,
+							TxidStr:     obfusTxID0,
+							OutputIndex: uint32(obfusOut0),
+						},
+					},
+				},
+			},
+		},
+		{
+			name:    "WalletBalance Response",
+			uri:     "/lnrpc.Lightning/WalletBalance",
+			msgType: rpcperms.TypeResponse,
+			msg: &lnrpc.WalletBalanceResponse{
+				TotalBalance:              1_000_000,
+				ConfirmedBalance:          1_000_000,
+				UnconfirmedBalance:        1_000_000,
+				LockedBalance:             1_000_000,
+				ReservedBalanceAnchorChan: 1_000_000,
+				AccountBalance: map[string]*lnrpc.WalletAccountBalance{
+					"first": {
+						ConfirmedBalance:   1_000_000,
+						UnconfirmedBalance: 1_000_000,
+					},
+				},
+			},
+
+			expectedReplacement: &lnrpc.WalletBalanceResponse{
+				TotalBalance:              950_100,
+				ConfirmedBalance:          950_100,
+				UnconfirmedBalance:        950_100,
+				LockedBalance:             950_100,
+				ReservedBalanceAnchorChan: 950_100,
+				AccountBalance: map[string]*lnrpc.WalletAccountBalance{
+					"first": {
+						ConfirmedBalance:   950_100,
+						UnconfirmedBalance: 950_100,
+					},
+				},
+			},
+		},
+		{
+			name:    "PendingChannels Response",
+			uri:     "/lnrpc.Lightning/PendingChannels",
+			msgType: rpcperms.TypeResponse,
+			msg: &lnrpc.PendingChannelsResponse{
+				PendingOpenChannels: []*lnrpc.PendingChannelsResponse_PendingOpenChannel{
+					{
+						CommitFee: 123,
+						Channel: &lnrpc.PendingChannelsResponse_PendingChannel{
+							RemoteNodePub: "01020304",
+							LocalBalance:  100_000,
+							RemoteBalance: 100_000,
+							Capacity:      120_000,
+							ChannelPoint: outPoint(
+								clearTxID, 0,
+							),
+							Initiator: lnrpc.Initiator_INITIATOR_LOCAL,
+							Memo:      "something",
+						},
+					},
+				},
+			},
+			expectedReplacement: &lnrpc.PendingChannelsResponse{
+				PendingOpenChannels: []*lnrpc.PendingChannelsResponse_PendingOpenChannel{
+					{
+						CommitFee: 123,
+						Channel: &lnrpc.PendingChannelsResponse_PendingChannel{
+							RemoteNodePub: "c8134495",
+							LocalBalance:  95100,
+							RemoteBalance: 95100,
+							Capacity:      114100,
+							ChannelPoint: outPoint(
+								obfusTxID0,
+								obfusOut0,
+							),
+							Initiator: lnrpc.Initiator_INITIATOR_LOCAL,
+							Memo:      "something",
+						},
+					},
+				},
+			},
+		},
+		{
+			name:    "ClosedChannels Response",
+			uri:     "/lnrpc.Lightning/ClosedChannels",
+			msgType: rpcperms.TypeResponse,
```

Comment:
I think we definitely should add coverage here for a test with the new `ClearClosingTxIds` flag set!